admin_panel/exams/model.py

```diff
@@ -0,0 +1,84 @@
+from django.db import models
+from django.core.validators import MinValueValidator
+
+from admin_panel.organizations.model import Organization
+from utils.timestamp_mixin import TimestampMixin
+
+
+class Exam(TimestampMixin):
+    """
+    Core Exam model containing only the basic exam information.
+    Settings are managed at the batch level.
+    """
+    DIFFICULTY_CHOICES = [
+        ('Easy', 'Easy'),
+        ('Medium', 'Medium'),
+        ('Hard', 'Hard'),
+    ]
+
+    OBJECTIVE_CHOICES = [
+        ('Semester Exam', 'Semester Exam'),
+        ('Mock Exam', 'Mock Exam'),
+        ('Entrance Exam', 'Entrance Exam'),
+        ('Test Series', 'Test Series'),
+    ]
+
+    # Basic Details
+    name = models.CharField(
+        max_length=255,
+        help_text="Name of the exam"
+    )
+    objective = models.CharField(
+        max_length=50,
+        choices=OBJECTIVE_CHOICES,
+        help_text="Purpose of the exam"
+    )
+    duration = models.DurationField(
+        help_text="Duration of the exam"
+    )
+    difficulty = models.CharField(
+        max_length=20,
+        choices=DIFFICULTY_CHOICES,
+        help_text="Difficulty level of the exam"
+    )
+    department = models.CharField(
+        max_length=255,
+        help_text="Department conducting the exam"
+    )
+    semester = models.CharField(
+        max_length=50,
+        help_text="Semester for which exam is being conducted"
+    )
+    subject = models.CharField(
+        max_length=255,
+        help_text="Subject of the exam"
+    )
+    exam_code = models.CharField(
+        max_length=50,
+        unique=True,
+        help_text="Unique code for the exam"
+    )
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Add validation for exam code format</b></div>
<div id="fix">

The `exam_code` field might benefit from a validator to ensure consistent formatting. Consider adding a regex validator to enforce a specific pattern for exam codes.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -56,6 +56,8 @@
      exam_code = models.CharField(
          max_length=50,
          unique=True,
 +        validators=[RegexValidator(r'^[A-Z0-9-]+$',
 +                   'Exam code must contain only uppercase letters, numbers and hyphens')],
          help_text="Unique code for the exam"
      )
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/sleep-404/ExamEvoke/pull/8#issuecomment-2499580485>#1f7607</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged